test/examples/nqueens_solveall.jl

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) 2022 MiniZinc.jl contributors
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+# N-queens example
+# based on MiniZinc example nqueens.mzn
+# queen in column i is in row q[i]
+
+n = 8
+
+function init_model()
+    model = MOI.instantiate(
+        () -> MiniZinc.Optimizer{Int}("chuffed");
+        with_cache_type = Int,
+        with_bridge_type = Int,
+    )
+    MOI.set(model, MOI.RawOptimizerAttribute("model_filename"), "test.mzn")
+    q = MOI.add_variables(model, n)
+    MOI.add_constraint.(model, q, MOI.Interval(1, n))
+    MOI.add_constraint(model, MOI.VectorOfVariables(q), MOI.AllDifferent(n))
+    for op in (+, -)
+        f = MOI.Utilities.vectorize([op(q[i], i) for i in eachindex(q)])
+        MOI.add_constraint(model, f, MOI.AllDifferent(n))
+    end
+
+    return model, q
+end
+
+function check_result(
+    model,
+    q,
+    actual_count = 92,
+    termination_status = MOI.OPTIMAL,
+)
+    @test MOI.get(model, MOI.TerminationStatus()) === termination_status
+    res_count = MOI.get(model, MOI.ResultCount())
+    @test res_count == actual_count
+
+    for i in 1:res_count
+        q_sol = MOI.get(model, MOI.VariablePrimal(i), q)
+        @test allunique(q_sol)
+        @test allunique(q_sol .+ (1:n))
+        @test allunique(q_sol .- (1:n))
+    end
+    @test MOI.get(model, MOI.SolveTimeSec()) < 4.0
+    return rm("test.mzn")
+end
+
+function run_nqueens(option)
```

Comment:
Split this function into separate `test_nqueens_xxx()` functions for each case so that we don't dispatch based on string input.